app/components/CopyButton/index.js

```diff
@@ -14,25 +15,38 @@ export default class CopyButton extends Component {
     className: '',
   };
 
+  static contextType = I18nContext;
+
   state = {
     hasCopied: false,
   };
 
+  timer = null;
+
+  componentWillUnmount() {
+    if (this.timer) clearTimeout(this.timer);
+  }
+
   copyAddress = () => {
     copy(this.props.content);
     this.setState({ hasCopied: true });
-    setTimeout(() => this.setState({ hasCopied: false }), 2500);
+    this.timer = setTimeout(() => {
+      this.setState({ hasCopied: false });
+      this.timer = null;
+    }, 2500);
   };
 
   render() {
+    const {t} = this.context;
+
     return (
       <button
         className={c('copy-btn', this.props.className, {
           'copy-btn--copied': this.state.hasCopied,
         })}
         onClick={this.copyAddress}
       >
-        Copy
+        {this.state.hasCopied ? `${t('copied')}!` : t('copy')}
```

Comment:
`!` should be part of the translation.

```suggestion
        {this.state.hasCopied ? t('copied') : t('copy')}
```